addons/account/tests/test_account_move_entry.py

```diff
@@ -1102,12 +1102,31 @@ def test_account_root_multiple_companies(self):
             'company_id': self.env.company.id,
         })
         other_company = self.env['res.company'].create({'name': 'other company'})
-        self.env['account.account'].create({
+        other_account = self.env['account.account'].create({
             'name': 'other account',
             'code': 'ZZ',
             'account_type': 'asset_current',
             'company_id': other_company.id,
         })
+
+        # ids are company-scoped
+        self.assertTrue(account.root_id, "root_id should be set for 2-char code")
+        self.assertTrue(other_account.root_id, "root_id should be set for 2-char code (other company)")
+        self.assertNotEqual(
+            account.root_id.id, other_account.root_id.id,
+            "Same code in different companies must yield different account.root ids",
+        )
+        # Check that the root is correctly set
+        zz_root = self.env['account.root'].with_company(self.env.company).search([('name', '=', 'ZZ')], limit=1)
+        self.assertTrue(zz_root, "Missing 'ZZ' root in current company")
+        self.assertEqual(zz_root.id, account.root_id.id)
+        self.assertTrue(zz_root.parent_id, "Two-digit root should have a one-digit parent")
+        self.assertEqual(zz_root.parent_id.name, 'Z')
+        self.assertEqual(zz_root.parent_id.company_id, account.company_id)
+        # No duplicate ids for a given company
+        ids = self.env['account.root'].with_company(self.env.company).search([]).ids
+        self.assertEqual(len(ids), len(set(ids)), "account.root search should not return duplicate ids")
+
```

Comment:
I think we should just check the id values explicitly. (Maybe some of the `name` or `company_id` checks are also useful; but we can just "extend" the assertion by using `assertRecordValues`)
```suggestion
        self.assertEqual(account.root_id.id, int(f'{self.env.company.id}122122'))
        self.assertEqual(other_account.root_id.id, int(f'{other_company.id}122122'))
        self.assertEqual(account.root_id.parent_id.id, int(f'{self.env.company.id}000122'))
        self.assertEqual(other_account.root_id.parent_id.id, int(f'{other_company.id}000122'))
```
(I used `zz` instead of `ZZ` to get 3 digit `ASCII` values)